src/precomputed_quantities.jl

```diff
@@ -3,49 +3,124 @@
 #####
 
 import Thermodynamics as TD
-import ClimaCore.Fields as Fields
-import ClimaCore.Geometry as Geometry
+import ClimaCore: Geometry, Spaces, Fields
 
-function precomputed_quantities!(Y, p, t)
-    Fields.bycolumn(axes(Y.c)) do colidx
-        precomputed_quantities!(Y, p, t, colidx)
-    end
-end
-
-function set_boundary_velocity!(uₕ, uᵥ)
-    uₕ_surface_data = Fields.level(Fields.field_values(uₕ), 1)
-    uₕ_surface_geom = Fields.level(Spaces.local_geometry_data(axes(uₕ)), 1)
-    uᵥ_surface_data = Fields.level(Fields.field_values(uᵥ), 1)
-    uᵥ_surface_geom = Fields.level(Spaces.local_geometry_data(axes(uᵥ)), 1)
+"""
+    set_boundary_velocity!(Y)
 
+Modifies `Y.f.w` so that `ᶠu³` is 0 at the surface.
+"""
+function set_boundary_velocity!(Y)
+    uₕ_surface_data = Fields.level(Fields.field_values(Y.c.uₕ), 1)
+    uₕ_surface_geom = Fields.level(Spaces.local_geometry_data(axes(Y.c.uₕ)), 1)
+    uᵥ_surface_data = Fields.level(Fields.field_values(Y.f.w), 1)
+    uᵥ_surface_geom = Fields.level(Spaces.local_geometry_data(axes(Y.f.w)), 1)
     @. uᵥ_surface_data = Geometry.Covariant3Vector(
         -Geometry.contravariant3(uₕ_surface_data, uₕ_surface_geom) /
         Geometry.contravariant3(one(uᵥ_surface_data), uᵥ_surface_geom),
     )
 end
 
-function precomputed_quantities!(Y, p, t, colidx)
-    ᶜuₕ = Y.c.uₕ
-    ᶠw = Y.f.w
-
-    (; ᶜK, ᶠu_tilde, ᶜu_bar, ᶠu³, ᶜts, ᶜp, params) = p
-    (; ᶠwinterp, ᶜinterp) = p.operators
-    ᶜJ = Fields.local_geometry_field(axes(Y.c.ρ)).J
-    C123 = Geometry.Covariant123Vector
-
-    set_boundary_velocity!(ᶜuₕ[colidx], ᶠw[colidx])
-
-    @. ᶜu_bar[colidx] = C123(ᶜuₕ[colidx]) + C123(ᶜinterp(ᶠw[colidx]))
-    @. ᶠu_tilde[colidx] =
-        ᶠwinterp(
-            Y.c.ρ[colidx] * ᶜJ[colidx],
-            Geometry.Contravariant123Vector(ᶜuₕ[colidx]),
-        ) + Geometry.Contravariant123Vector(ᶠw[colidx])
-    @. ᶠu³[colidx] =
-        Geometry.project(Geometry.Contravariant3Axis(), ᶠu_tilde[colidx])
-    compute_kinetic!(ᶜK[colidx], ᶜuₕ[colidx], ᶠw[colidx])
-    thermo_params = CAP.thermodynamics_params(params)
-    thermo_state!(Y, p, ᶜinterp, colidx; time = t)
-    @. ᶜp[colidx] = TD.air_pressure(thermo_params, ᶜts[colidx])
-    return nothing
+"""
+    set_velocity_quantities!(ᶜu, ᶠu³, ᶜK, ᶠw, ᶜYc)
+
+Uses `ᶠw` and `ᶜYc` to set the values of `ᶜu`, `ᶠu³`, and `ᶜK`, assuming that
+the horizontal velocity is `ᶜYc.uₕ` and that it should be interpolated to cell
+faces using `ᶜYc.ρ * ᶜJ` in the weighted interpolation.
+"""
+function set_velocity_quantities!(ᶜu, ᶠu³, ᶜK, ᶠw, ᶜYc)
```

Comment:
So that the SGS velocity quantities can be computed using the same function, but with `ᶠsgsʲs[j].w` instead of `ᶠw` (still using the same `ᶜYc`, though).